cdap-security/src/main/java/co/cask/cdap/security/server/ExternalAuthenticationServer.java

```diff
@@ -173,12 +173,31 @@ protected void startUp() throws Exception {
         if (keyPassword != null && keyPassword.length() != 0) {
           sslContextFactory.setKeyManagerPassword(keyPassword);
         }
+
+        if (configuration.getBoolean(Constants.Security.CLIENT_AUTH_ENABLED, false)) {
+          String trustStorePath = configuration.get(Constants.Security.AuthenticationServer.SSL_TRUSTSTORE_PATH);
+          String trustStorePassword = configuration.get(Constants.Security.AuthenticationServer.SSL_TRUSTSTORE_PASSWORD);
+          String trustStoreType = configuration.get(Constants.Security.AuthenticationServer.SSL_TRUSTSTORE_TYPE,
+                  Constants.Security.AuthenticationServer.DEFAULT_SSL_KEYSTORE_TYPE);
+
+
+          Preconditions.checkArgument(trustStorePath != null, "Trust Store Path Not Configured");
+
+          /* SSL handshaking will involve requesting for a client certificate, if cert is not provided
+           * server continues with the connection but the client is considered to be unauthenticated */
+          sslContextFactory.setWantClientAuth(true);
```

Comment:
Our style is that comments such as these start with "//"; note minor corrections to text:
```
+          // SSL handshaking will involve requesting a client certificate; if a cert is not provided, the
+          // server continues with the connection but the client is considered to be unauthenticated